### ISSUE TITLE
Consider button focus rings on immediate child

### ISSUE THREAD

**Issue by danirabbit** [elementary/granite#807]
## Problem

We have some issues with focus rings being outside of buttons like clipping in linked styles or when there isn’t technically any margin around the button or some issue with z axis ordering or clipping in overlays. Lots of little edge cases that make focus rings outside of the button break

## Proposal

Maybe instead, we should apply the focus style to the immediate child of the button (likely a label or icon) as a background 

## Prior Art

This would look more similar to the flat button focus styles and indeed maybe we could combine a lot of the focus style code this way as well

**Comment by jeremypw** [comment]
@danirabbit Are there plans to progress this? A simple solution would be to focus the Gtk.Switch element only as that already has visible focus rings. I have tried this in settings-network and it works well.  The switch works normally with a clicks but is accessible with the keyboard.

**Comment by danirabbit** [comment]
@jeremypw yes we're reworking focus styles as part of the new granite stylesheet
